runtime/services/catalog/migrator/models/models.go

```diff
@@ -59,8 +60,39 @@ func (m *modelMigrator) Delete(ctx context.Context, olap drivers.OLAPStore, cata
 	})
 }
 
-func (m *modelMigrator) GetDependencies(ctx context.Context, olap drivers.OLAPStore, catalog *drivers.CatalogEntry) []string {
-	return ExtractTableNames(catalog.GetModel().Sql)
+func (m *modelMigrator) GetDependencies(ctx context.Context, olap drivers.OLAPStore, catalog *drivers.CatalogEntry) ([]string, []*drivers.CatalogEntry) {
+	model := catalog.GetModel()
+	model.Sql = sanitizeQuery(model.Sql)
+	dependencies := ExtractTableNames(model.Sql)
+
+	embeddedSourcesMap := make(map[string]*drivers.CatalogEntry)
+	for i, dependency := range dependencies {
+		source, ok := sources.GetEmbeddedSource(dependency)
+		if !ok {
+			continue
+		}
+		if _, ok := embeddedSourcesMap[source.Name]; ok {
+			continue
+		}
+
+		embeddedSourcesMap[source.Name] = &drivers.CatalogEntry{
+			Name:     source.Name,
+			Type:     drivers.ObjectTypeSource,
+			Object:   source,
+			Path:     source.Properties.AsMap()["path"].(string),
+			Embedded: true,
+		}
+
+		// replace the dependency
+		dependencies[i] = source.Name
+		model.Sql = strings.ReplaceAll(model.Sql, dependency, source.Name)
```

Comment:
This replacement makes me a little nervous since it can also impact other strings. E.g. in:
```
select *, 'Found in file "s3://bucket/file.csv"' as log from "s3://bucket/file.csv"
```
A quick way to make it safer would be to do the replace on the full original regex match (e.g. `FROM "s3://bucket/file.csv"` including whitespace is replaced to `FROM generated_name`). Could be done by returning a complex result from `ExtractTableNames`.

Not sure if worth it, let me know what you think

Reply:
I am not sure how much this case is encountered. One way to do it is to replace the new name in all the matches. The right way to do it using parsers. @magorlick @hamilton any thoughts?

Reply:
No strong feelings; I see hte argument in both directions. In the interest of time, I'd like to get this merged for v20 so I think what we have is acceptable for now. If someone encounters a bug by doing this, we'll address then. I doubt we'll need to before we have a SQL parser.